pilot/pkg/networking/core/v1alpha3/listener_builder.go

```diff
@@ -447,28 +444,64 @@ func newInboundPassthroughFilterChains(configgen *ConfigGeneratorImpl,
 			Push:             push,
 			ListenerProtocol: plugin.ListenerProtocolTCP,
 		}
+		var allChains []plugin.FilterChain
+		for _, p := range configgen.Plugins {
+			chains := p.OnInboundPassthroughFilterChains(in)
+			allChains = append(allChains, chains...)
+		}
+
+		if len(allChains) == 0 {
+			// Add one empty entry to the list if none of the plugins are interested in updating the filter chains.
+			allChains = []plugin.FilterChain{{}}
+		}
+		// Override the filter chain match to make sure the pass through filter chain captures the pass through traffic.
+		for i := range allChains {
+			chain := &allChains[i]
+			if chain.FilterChainMatch == nil {
+				chain.FilterChainMatch = &listener.FilterChainMatch{}
+			}
+			// Port : EMPTY to match all ports
+			chain.FilterChainMatch.DestinationPort = nil
+			chain.FilterChainMatch.PrefixRanges = []*core.CidrRange{
+				util.ConvertAddressToCidr(matchingIP),
+			}
+			chain.ListenerProtocol = plugin.ListenerProtocolTCP
+		}
+
 		mutable := &plugin.MutableObjects{
-			FilterChains: []plugin.FilterChain{
-				{
-					ListenerProtocol: plugin.ListenerProtocolTCP,
-				},
-			},
+			FilterChains: allChains,
 		}
 		for _, p := range configgen.Plugins {
 			if err := p.OnInboundPassthrough(in, mutable); err != nil {
 				log.Errorf("Build inbound passthrough filter chains error: %v", err)
 			}
 		}
 
-		filterChain := &listener.FilterChain{
-			FilterChainMatch: &filterChainMatch,
-			Filters:          append(mutable.FilterChains[0].TCP, tcpProxyFilter),
+		// Construct the actual filter chains for each of the filter chain from the plugin.
+		for _, chain := range allChains {
+			filterChain := &listener.FilterChain{
+				FilterChainMatch: chain.FilterChainMatch,
+				Filters:          append(chain.TCP, tcpProxyFilter),
+			}
+			if chain.TLSContext != nil {
```

Comment:
nit: can't this just be 

`TransportSocket: chain.TLSContext` in the line above?

Reply:
Hmm, I'm not sure will it cause any problems if `chain.TLSContext` is nil in this case?

Reply:
nevermind, I misunderstood this